### PR TITLE
[FIXED] update of R1 Consumer in clustered setup.

### PR DESCRIPTION
**Pull request by matthiashanel**
missing reply caused timeout

Signed-off-by: Matthias Hanel <mh@synadia.com>

 - [ ] Link to issue, e.g. `Resolves #NNN`
 - [ ] Documentation added (if applicable)
 - [ ] Tests added
 - [ ] Branch rebased on top of current main (`git pull --rebase origin main`)
 - [ ] Changes squashed to a single commit (described [here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
 - [ ] Build is green in Travis CI
 - [ ] You have certified that the contribution is your original work and that you license the work to the project under the [Apache 2 license](https://github.com/nats-io/nats-server/blob/main/LICENSE)

Resolves #

### Changes proposed in this pull request:

 -
 -
 -

/cc @nats-io/core

### CODE REVIEW

**Review by derekcollison — APPROVED** [review]
LGTM! Thanks.

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -2915,25 +2915,30 @@ func (js *jetStream) processClusterCreateConsumer(ca *consumerAssignment, state
 			}
 		}
 
+		update := false
 		// Start our monitoring routine.
 		if rg.node == nil {
 			// Single replica consumer, process manually here.
 			js.processConsumerLeaderChange(o, true)
+			update = true
```
Comment (kozlovic) [inline server/jetstream_cluster.go:2923]: But processConsumerLeaderChange() will send something out (see bottom of that function). So it feels like you are going to send 2 responses in some cases?

**Review by kozlovic — COMMENTED** [review]
Not sure if this is correct...

File: server/jetstream_cluster.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_cluster.go:2923]: Good catch.. hmm..

Comment (derekcollison) [reply server/jetstream_cluster.go:2923]: Need to clear ca.responded but that has to be under js lock.

https://github.com/nats-io/nats-server/blob/c36405ffbc05680d44e8bd75f3a480f1fe2a25ef/server/jetstream_cluster.go#L3345

Comment (derekcollison) [reply server/jetstream_cluster.go:2923]: Meaning clear it here before calling into processConsumerLeaderChange..

Comment (matthiashanel) [reply server/jetstream_cluster.go:2923]: basically ca.responded is never set to false, which is why it is not sent

Comment (matthiashanel) [reply server/jetstream_cluster.go:2923]: what's the use of this value even? shouldn't that be part of the stack, calling context?

Comment (matthiashanel) [reply server/jetstream_cluster.go]: Implemented it as Derek suggested. but really, not sure why this variable needs to exist in the first place. 
Shouldn't we always respond when called via `applyMetaEntries` and never when not?

Comment (derekcollison) [reply server/jetstream_cluster.go]: Its purpose is to not produce responses after we have already responded.

File: server/jetstream_cluster.go [hunk 2]
```diff
@@ -2918,6 +2918,9 @@ func (js *jetStream) processClusterCreateConsumer(ca *consumerAssignment, state
 		// Start our monitoring routine.
 		if rg.node == nil {
 			// Single replica consumer, process manually here.
+			js.mu.Lock()
+			ca.responded = false
```
Comment (derekcollison) [inline server/jetstream_cluster.go:2922]: Maybe add quick comment, // to force response in case we think we have responded before.

**Review by derekcollison — APPROVED** [review]
LGTM, small ask for comment.

**Comment by derekcollison** [comment]
Merge when ready. Thx

**Pull request merged**